### PR TITLE
Dashboard duplication support

### PR DESCRIPTION
**Pull request by jp-harvey**
Porting some code I wrote recently for duplicating dashboards into pymapd. Also has an experimental option to swap out the underlying dashboard datasources on duplication.

### CODE REVIEW

File: pymapd/_transforms.py [hunk 1]
```diff
@@ -0,0 +1,67 @@
+import json
+from base64 import b64decode, b64encode
+
+
+def change_dash_sources(dashboard, remap):
```
Comment (randyzwitch) [inline pymapd/_transforms.py]: @jp-harvey can you write some brief comments about what each part of these chunks do? I get that you are parsing through the JSON object and changing things, but it's hard to follow. It would also be great if the variables had more descriptive names and counter variables were unique (reusing `i` within the same outer loop is confusing)

File: pymapd/_transforms.py [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+                            ['table']
+                         ) = nt
+                    im += 1
+                il += 1
+        (ds
+            ['dashboard']
+            ['dataSources']
+            [nt]
+         ) = ds['dashboard']['dataSources'].pop(ot)
+    ds = b64encode(json.dumps(ds).encode()).decode()
```
Comment (randyzwitch) [inline pymapd/_transforms.py]: For ds and dm, feels like you should just do the assignment to `d.dashboard_state` and `d.dashboard_metadata` directly, no need for separate lines`

File: pymapd/_transforms.py [hunk 3]
```diff
@@ -0,0 +1,67 @@
+import json
+from base64 import b64decode, b64encode
+
+
+def change_dash_sources(dashboard, remap):
+    d = dashboard
```
Comment (randyzwitch) [inline pymapd/_transforms.py]: Could you just use `dashboard` and `remap` within the code, so it makes it more obvious that these are the values passed in through the function arguments?

File: pymapd/_transforms.py [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    r = remap
+    dm = json.loads(d.dashboard_metadata)
+    tlst = map(str.strip, dm.get('table', '').split(','))
+    tlst = [r[t]['name'] if r.get(t, {}).get('name', {}) else t for t in tlst]
```
Comment (randyzwitch) [inline pymapd/_transforms.py]: Between all these code lines, it'd be great to add a new line between separate operations so it's clear a new operation is occurring (especially around the table renaming section and loops)

File: pymapd/_transforms.py (hunk 3 above)
Comment (jp-harvey) [reply pymapd/_transforms.py]: Yes, but it makes it much more verbose and splits many things over multiple lines due to Flake8 requirements, making it less readable

**Comment by randyzwitch** [comment]
Will review Wednesday since this now passes

**Comment by randyzwitch** [comment]
@Zeerg Merging this since @jp-harvey indicated that it's time sensitive, but please plan on adding a test in the near future. If you need any help, please feel free to send me a Slack message.
